[DROOLS-5910] Enable range index for JoinNode (#3606)
kiegroup/kie-docs: doc-content/drools-docs/src/main/asciidoc/DecisionEngine/performance-tuning-decision-engine-ref.adoc

Alpha node range index is used to evaluate the rule constraint. You can configure the threshold of the alpha node range index using the `drools.alphaNodeRangeIndexThreshold` system property. The default value of the threshold is `9`, indicating that the alpha node range index is enabled when a precedent node contains more than nine alpha nodes with inequality constraints. For example, when you have nine rules similar to `Person(age > 10)`, `Person(age > 20)`, ..., `Person(age > 90)`, then you can have similar nine alpha nodes.
+
The default value of the threshold is based on the related advantages and overhead. However, if you configure a smaller value for the threshold, then the performance can be improved depending on your rules. For example, you can configure the `drools.alphaNodeRangeIndexThreshold` value to `6`, enabling the alpha node range index when you have more than six alpha nodes for a precedent node. You can set a suitable value for the threshold based on the performance test results of your rules.

Enable join node range index::

> Join node range index is implemented but it's disabled by default because it gives the performance improvement only when there is a large number of facts to be joined, for example, 256*16 combinations. If your application inserts such a large number of facts, you can enable join node range index and evaluate the performance improvement.

Reviewer: ```suggestion
The join node range index feature improves the performance only when there is a large number of facts to be joined, for example, 256*16 combinations. When your application inserts a large number of facts, you can enable the join node range index and evaluate the performance improvement. By default, the join node range index is disabled. 
```

+
--
.Example `kmodule.xml` file
[source,xml]
----
<kbase name="KBase1" betaRangeIndex="enabled">
----

> .System property for BetaRangeIndexOption

Reviewer: ```suggestion
.System property for `BetaRangeIndexOption`
```
